### ISSUE TITLE
Stop Auto-sklearn if an ensemble error happens

### ISSUE THREAD

**Issue by mfeurer** [automl/auto-sklearn#581]
Because this renders the results useless.

**Comment by kaiweiang** [comment]
@mfeurer can I know what type of ensemble errors that are normally seen?
Currently, I am seeing autosklearn model I ran using a custom optimization metric has multiple successful runs but none of them was selected as part of the final ensemble. Instead, dummyclassifier was selected. Was there any error occurred when ensembling?

**Comment by kaiweiang** [comment]
@mfeurer It appears the root cause was from the recent fix #522 . This line of code 
```sorted_keys = filter(lambda x: x[1] > dummy_score[1], sorted_keys)``` 
removes all the runs that have lower score than the dummy_score. The custom metric that I passed gives a relatively high dummy_score thus causing all the good runs get filtered out.

**Comment by mfeurer** [comment]
Is your metric a loss function which should be minimized? Otherwise, if the dummy_score is higher it means that your models are worse than doing a constant predictions of the majority class. Would it be possible that you provide code to reproduce your issue?

**Comment by kaiweiang** [comment]
The custom metric I use is for binary classification. It is higher the better. It highly favors the recall for positive class. I do realize the dummyclassifier always predicts zero only, hence, if my positive class is zero, I will get a very high score as the dummy_score. Any idea why dummyclassifier always predicts zero, because I think it is using the uniform strategy and it should predict one also?

**Comment by kaiweiang** [comment]
If my positive class is one, it wouldn't have issues since the dummyclassifier always predict zero.

**Comment by mfeurer** [comment]
>  Any idea why dummyclassifier always predicts zero, because I think it is using the uniform strategy and it should predict one also?

No, sorry, I don't know. You either need to share a minimal example to reproduce this or debug it yourself.

> If my positive class is one, it wouldn't have issues since the dummyclassifier always predict zero.

I think it would be very good if you use one as the positive class as this is quite standard.

**Comment by kaiweiang** [comment]
Sure, but currently my pc is not with me. I'll try to provide an example code asap.
Thanks

**Comment by franchuterivera** [comment]
Autosklearn now is robust to these problems (that is failures in the ensemble selection) through the use of dask multiple calls. I am closing this issue as it is no longer a problem now.
